static/app/views/issueDetails/streamline/useSuspectFlags.tsx

```diff
@@ -0,0 +1,116 @@
+import moment from 'moment-timezone';
+
+import type {Event} from 'sentry/types/event';
+import type {Organization} from 'sentry/types/organization';
+import {trackAnalytics} from 'sentry/utils/analytics';
+import {useApiQuery} from 'sentry/utils/queryClient';
+
+type RawFlag = {
+  action: string;
+  created_at: string;
+  created_by: string;
+  created_by_type: string;
+  flag: string;
+  id: number;
+  tags: Record<string, any>;
+};
+
+export type RawFlagData = {data: RawFlag[]};
+
+export type FlagSeriesDatapoint = {
+  // flag action
+  label: {formatter: () => string};
+  // flag name
+  name: string;
+  // unix timestamp
+  xAxis: number;
+};
+
+export function hydrateFlagData({
+  rawFlagData,
+}: {
+  rawFlagData: RawFlagData | undefined;
+}): FlagSeriesDatapoint[] {
+  if (!rawFlagData) {
+    return [];
+  }
+
+  // transform raw flag data into series data
+  // each data point needs to be type FlagSeriesDatapoint
+  const flagData = rawFlagData.data.map(f => {
+    return {
+      xAxis: Date.parse(f.created_at),
+      label: {formatter: () => f.action},
+      name: `${f.flag}`,
+    };
+  });
+  return flagData;
+}
+
+export default function useSuspectFlags({
+  organization,
+  query,
+  firstSeen,
+  rawFlagData,
+  event,
+}: {
+  event: Event;
+  firstSeen: string;
+  organization: Organization;
+  query: Record<string, any>;
+  rawFlagData: RawFlagData | undefined;
+}) {
+  const hydratedFlagData: FlagSeriesDatapoint[] = hydrateFlagData({rawFlagData});
```

Comment:
It is already typed by the return type, so no need to type it :)
```suggestion
  const hydratedFlagData = hydrateFlagData(rawFlagData);
```